### PR TITLE
10176 - Optimize Speed fetching Pending Motions

### PR DESCRIPTION
**Pull request by cruzjone-flexion**

### CODE REVIEW

File: web-api/src/persistence/elasticsearch/docketEntry/getAllDocketEntries.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import { MOTION_EVENT_CODES } from '@shared/business/entities/EntityConstants';
+import { QueryDslQueryContainer } from '@opensearch-project/opensearch/api/types';
+import { calculateISODate } from '@shared/business/utilities/DateHandler';
+import { pendingItemCaseSource } from '@web-api/business/useCases/pendingItems/fetchPendingItemsInteractor';
+import { searchAll } from '@web-api/persistence/elasticsearch/searchClient';
+
+export const getAllDocketEntries = async ({
```
Comment (nechama-krigsman) [inline web-api/src/persistence/elasticsearch/docketEntry/getAllDocketEntries.ts]: Can this method either be renamed to indicate that it fetches the pending motions that are older than 180 days or take those filters in as parameters?

File: shared/src/business/useCases/pendingMotion/getPendingMotionDocketEntriesForCurrentJudgeInteractor.ts [hunk 2]
```diff
@@ -35,51 +34,52 @@ export const getPendingMotionDocketEntriesForCurrentJudgeInteractor = async (
...
-        applicationContext,
-        docketNumber,
-        includeConsolidatedCases: true,
+  const { results: allDocketEntries } = await applicationContext
+    .getPersistenceGateway()
+    .getAllPendingMotionDocketEntriesForJudge({ applicationContext, judge });
+
+  const currentDate = prepareDateFromString().toISO()!;
+
+  const batchCases = (
```
Comment (nechama-krigsman) [inline shared/src/business/useCases/pendingMotion/getPendingMotionDocketEntriesForCurrentJudgeInteractor.ts:43]: Are these still batchCases?

File: shared/src/business/useCases/pendingMotion/getPendingMotionDocketEntriesForCurrentJudgeInteractor.ts [hunk 3]
```diff
@@ -35,51 +34,52 @@ export const getPendingMotionDocketEntriesForCurrentJudgeInteractor = async (
...
+        const latestDocketEntry: RawDocketEntry = fullCase.docketEntries.find(
+          de => de.docketEntryId === docketEntry.docketEntryId,
+        );
+
+        const dayDifference = calculateDifferenceInDays(
+          currentDate,
+          docketEntry.createdAt,
+        );
+
+        const updatedDocketEntry: RawDocketEntry & {
```
Comment (nechama-krigsman) [inline shared/src/business/useCases/pendingMotion/getPendingMotionDocketEntriesForCurrentJudgeInteractor.ts:63]: Does this include caseCaption and leadDocketNumber?

**Review by nechama-krigsman — APPROVED** [review]

File: web-api/src/persistence/elasticsearch/docketEntry/getAllDocketEntries.ts (hunk 1 above)
Comment (cruzjone-flexion) [reply web-api/src/persistence/elasticsearch/docketEntry/getAllDocketEntries.ts]: I think that is a stale file, i did switch it to "getAllPendingMotionDocketEntriesForJudge" file name and method name

File: shared/src/business/useCases/pendingMotion/getPendingMotionDocketEntriesForCurrentJudgeInteractor.ts (hunk 2 above)
Comment (cruzjone-flexion) [reply shared/src/business/useCases/pendingMotion/getPendingMotionDocketEntriesForCurrentJudgeInteractor.ts:43]: No, I will be refactoring once deployed to get better naming and typing

**Pull request merged**